cmd/kubeadm/app/node/bootstrap.go

```diff
@@ -128,15 +128,16 @@ func checkForNodeNameDuplicates(clientSet *clientset.Clientset) error {
 	if err != nil {
 		return fmt.Errorf("Failed to get node hostname [%v]", err)
 	}
-	nodeList, err := clientSet.Nodes().List(v1.ListOptions{})
+	node, err := clientSet.Nodes().Get(hostName, v1.GetOptions{})
```

Comment:
@liggitt suggested that getting the node is a safer than listing nodes in regards to RBAC permissions.